e2e/network_test.go

```diff
@@ -0,0 +1,310 @@
+		t.Fatal(result.Error())
+	}
+	defer func() {
+		uri = fmt.Sprintf("/apis/%s/%s/namespaces/%s/%s/%s",
+			strings.ToLower("extensions"),
+			strings.ToLower("v1beta1"),
+			strings.ToLower(namespace),
+			strings.ToLower("NetworkPolicies"),
+			strings.ToLower(np.ObjectMeta.Name))
+
+		result = httpRestClient.Delete().RequestURI(uri).Do()
+		if result.Error() != nil {
+			t.Fatal(result.Error())
+		}
+
+	}()
+
+	//
+	//
+	//
+	// 6. create a wget pod with label `allow=access` that hits the nginx service
+	testPodName := fmt.Sprintf("%s-%s", "wget-pod-nt", utilrand.String(5))
+	wgetPodNT.ObjectMeta.Name = testPodName
+	wgetPodNT.ObjectMeta.Labels = map[string]string{}
+	wgetPodNT.ObjectMeta.Labels["allow"] = "access"
+	_, err = client.CoreV1().Pods(namespace).Create(wgetPodNT)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := retry(10, time.Second*10, getPod(testPodName)); err != nil {
+		t.Fatalf(fmt.Sprintf("timed out waiting for wget pod to succeed: %v", err))
+	}
+
+	//
+	//
+	//
+	// 7. create a wget pod with label `allow=cant-access` that fails to the nginx service
+	testPodName = fmt.Sprintf("%s-%s", "wget-pod-nt", utilrand.String(5))
+	wgetPodNT.ObjectMeta.Name = testPodName
+	wgetPodNT.ObjectMeta.Labels["allow"] = "cant-access"
+	_, err = client.CoreV1().Pods(namespace).Create(wgetPodNT)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := retry(10, time.Second*10, getFailedPod(testPodName)); err != nil {
+		t.Fatalf(fmt.Sprintf("timed out waiting for wget pod to fail: %v", err))
+	}
+}
+
+func getNginxPod() error {
+	l, err := client.CoreV1().Pods(namespace).List(metav1.ListOptions{LabelSelector: "app=nginx"})
+	if err != nil || len(l.Items) == 0 {
+		return fmt.Errorf("pod not yet running: %v", err)
+	}
+
+	// take the first pod
+	p := &l.Items[0]
+
+	if p.Status.Phase != v1.PodRunning {
+		return fmt.Errorf("pod not yet running: %v", p.Status.Phase)
+	}
+	return nil
+}
+
+func getPod(name string) func() error {
+	return func() error {
+		p, err := client.CoreV1().Pods(namespace).Get(name, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("pod not yet running: %v", err)
+		}
+		if p.Status.Phase != v1.PodSucceeded {
+			return fmt.Errorf("pod not yet running: %v", p.Status.Phase)
+		}
+		return nil
+	}
+}
+
+func getFailedPod(name string) func() error {
+	return func() error {
+		p, err := client.CoreV1().Pods(namespace).Get(name, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("pod not yet running: %v", err)
+		}
+		if p.Status.Phase != v1.PodFailed {
+			return fmt.Errorf("pod not yet running: %v", p.Status.Phase)
+		}
+		return nil
+	}
+}
+
+var nginxDepNT = []byte(`apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: nginx-deployment-nt
+spec:
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.8
+        ports:
+        - containerPort: 80
+`)
+
+var wgetPodNT = &v1.Pod{
```

Comment:
wgetPodNT is used to start ping pods with different names so it is easier to change it from struct. While others are used as manifest files!